### PR TITLE
TORCHSCRIPT_CLASSIFIER node for any user-provided .torchscript AI model

### PR DESCRIPTION
**Pull request by Roulbac**
### Description

- [X] I've included a concise description of what each node does

### Styleguide

- [X] My node adheres to the [styleguide](https://github.com/flojoy-io/nodes/blob/main/node_styleguide.md) for Flojoy nodes

### Docs

- [X] I've submitted a PR for a documentation page for the new node(s) that contains usage examples (see docs.flojoy.io)

### Testing

- [X] This PR includes a unit test ([example here](https://github.com/flojoy-io/nodes/blob/c28808f2d0aec8f116cdb3fccb46c4f5256574e9/TRANSFORMERS/ARITHMETIC/ADD/ADD_test_.py) and/or ideally a screenshot of the node's output on an example app.

### CODE REVIEW

File: AI_ML/CLASSIFICATION/TORCHSCRIPT_CLASSIFIER/assets/President_Barack_Obama.jpg
Comment (jjerphan) [inline AI_ML/CLASSIFICATION/TORCHSCRIPT_CLASSIFIER/assets/President_Barack_Obama.jpg:1]: Not urgent, not related to this PR: probably we should have shared `assets` for Node at a point.

File: AI_ML/CLASSIFICATION/TORCHSCRIPT_CLASSIFIER/assets/mbnet_v3_small.torchscript
Comment (jjerphan) [inline AI_ML/CLASSIFICATION/TORCHSCRIPT_CLASSIFIER/assets/mbnet_v3_small.torchscript:1]: I think this model is relatively large as a model to be stored as an asset in a git repository.

If this serialized model is published somewhere, could we have the `TORCHSCRIPT_CLASSIFIER` node support URL as well?

Alternatively, if we want to store large artifacts, we might want to have a look at git LFS but this might come with other usability problems and friction for development.

What do you think?

**Review by jjerphan — COMMENTED** [review]
This looks like a valuable contribution to me.

To review this PR, I locally updated this branch merging `develop` (now used as a base branch), performed a few required changes, and pushed it on my fork for now: https://github.com/jjerphan/nodes/commits/reda-torchscript-classifier/).

Note related to this PR: we might want to take some time to think of reorganizing the `AI_ML` section.

I have a few comments and questions.

File: AI_ML/CLASSIFICATION/ONE_HOT_ENCODING/ONE_HOT_ENCODING.py
[the remark this answers is not in the document]
Comment (Roulbac) [reply AI_ML/CLASSIFICATION/ONE_HOT_ENCODING/ONE_HOT_ENCODING.py:1]: The PRs must have been stacked in the past, fixing that atm thanks for the heads up

File: AI_ML/CLASSIFICATION/TORCHSCRIPT_CLASSIFIER/assets/President_Barack_Obama.jpg
Comment (Roulbac) [reply AI_ML/CLASSIFICATION/TORCHSCRIPT_CLASSIFIER/assets/President_Barack_Obama.jpg:1]: Actually this is a great idea let's do it. It's time we have some more shared assets and fixtures

File: AI_ML/CLASSIFICATION/TORCHSCRIPT_CLASSIFIER/assets/mbnet_v3_small.torchscript
Comment (Roulbac) [reply AI_ML/CLASSIFICATION/TORCHSCRIPT_CLASSIFIER/assets/mbnet_v3_small.torchscript:1]: Yeah that's true... I think however that 10Mb is ok, this model comes from the torchvision zoo and I know they have auto rate limiting which can be problematic with CI. If we run into serious problems it might then be worth making it smaller.

File: AI_ML/CLASSIFICATION/ONE_HOT_ENCODING/ONE_HOT_ENCODING.py
[the remark this answers is not in the document]
Comment (Roulbac) [reply AI_ML/CLASSIFICATION/ONE_HOT_ENCODING/ONE_HOT_ENCODING.py:1]: Merged the previous one into `develop` and the issue is now fixed

File: AI_ML/CLASSIFICATION/TORCHSCRIPT_CLASSIFIER/app.txt
[the remark this answers is not in the document]
Comment (Roulbac) [reply AI_ML/CLASSIFICATION/TORCHSCRIPT_CLASSIFIER/app.txt:1]: Good idea, following that up shortly. I think it's however worth publishing in the gallery once we have a tutorial on how to bring a model and use it with this node end-to-end

File: AI_ML/CLASSIFICATION/TORCHSCRIPT_CLASSIFIER/assets/mbnet_v3_small.torchscript
Comment (jjerphan) [reply AI_ML/CLASSIFICATION/TORCHSCRIPT_CLASSIFIER/assets/mbnet_v3_small.torchscript:1]: I would be very cautious of vendoring large assets even if they are removed then as they still are still represented in git objects (hence the repository remains large) except if commits are entirely purged from them; this would necessitate rewriting the entire history starting from the commit which introduced them, which is hardly practicable, I think.

How about proceeding the other way around that is to try using Torchvision zoo first?

Even though we might hit rate limit, I think that there are ways to properly handle that, for instance:
 - on the short term, we could mark test `xfail` if the model can't be downloaded (due to network of rate limit)
 - on the medium term, we could have nightly CI and have an extended tests suite which would [run tests which necessitates network to prevent disruption using models (using all the nightly of dependencies) once a day](https://github.com/scikit-learn/scikit-learn/blob/b72252e9629d8cf618397f7a2ab4da28fa1021f6/azure-pipelines.yml#L64-L66)

Some project CI (as scikit-learn's for instance as linked above) adopted this approach, and it has significantly reduced frictions while properly testing new configurations as soon as available and preventing major outages before releases.

What do you think of this?

Comment (Roulbac) [reply AI_ML/CLASSIFICATION/TORCHSCRIPT_CLASSIFIER/assets/mbnet_v3_small.torchscript:1]: That makes sense.
Our current CI setup for heavy AI nodes (marked with `@pytest.mark.skip`) runs nightly there is no change needed.
Moving from `torchvision` to `torch.hub` should remove or reduce rate limiting errors from Github (see the new fixture)

File: AI_ML/CLASSIFICATION/TORCHSCRIPT_CLASSIFIER/TORCHSCRIPT_CLASSIFIER_test_.py [hunk 1]
```diff
@@ -0,0 +1,82 @@
...
+    # torch which is why we need to run this in a venv.
+    @run_in_venv(pip_dependencies=["torch~=2.0.1", "torchvision~=0.15.2"])
+    def _download_test_model(path: str):
+        import torch
+        import torchvision
+
+        class ModelWithTransform(torch.nn.Module):
+            def __init__(self):
+                super().__init__()
+                self.model = torch.hub.load(
```
Comment (jjerphan) [inline AI_ML/CLASSIFICATION/TORCHSCRIPT_CLASSIFIER/TORCHSCRIPT_CLASSIFIER_test_.py:22]: [`torch.hub`'s models' cache depending on the presence of the `TORCH_HOME` environment variable](https://pytorch.org/docs/stable/hub.html#where-are-my-downloaded-models-saved).

Should we do another PR similar to https://github.com/flojoy-ai/python/pull/97 but to be able to set `TORCH_HOME` to Flojoy's cache directory?

**Review by jjerphan — APPROVED** [review]
LGTM! Thank you, @Roulbac.

Just one last side-note.

I let you merge if when ready.

File: AI_ML/CLASSIFICATION/TORCHSCRIPT_CLASSIFIER/TORCHSCRIPT_CLASSIFIER_test_.py (hunk 1 above)
Comment (Roulbac) [reply AI_ML/CLASSIFICATION/TORCHSCRIPT_CLASSIFIER/TORCHSCRIPT_CLASSIFIER_test_.py:22]: Good point
